### PR TITLE
Initial attempt for ESM support

### PR DESCRIPTION
**Pull request by remcohaszing**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -72,8 +67,8 @@ if (electron && nvm && !fs.existsSync(globals)) {
  * @param {LoadPluginOptions} [options]
  * @returns {any}
  */
-function loadPlugin(name, options) {
-  return require(resolvePlugin(name, options) || name)
+export default function loadPlugin(name, options) {
```
Comment (remcohaszing) [inline index.js:70]: Tests rely on this being the default export. I personally prefer a named export, especially for packages that export multiple members.

File: index.js [hunk 2]
```diff
@@ -1,20 +1,15 @@
...
-var resolve = require('resolve-from').silent
+import fs from 'fs'
+import path from 'path'
+import {silent as resolve} from 'resolve-from'
 // type-coverage:ignore-next-line
-var readNpmConfig = require('libnpmconfig').read
-
-module.exports = loadPlugin
-loadPlugin.resolve = resolvePlugin
+import {read as readNpmConfig} from 'libnpmconfig'
```
Comment (remcohaszing) [inline index.js:12]: `libnpmconfig` relies on [`figgy-pudding`](https://github.com/npm/figgy-pudding), which is marked for deprecation. I think it’s better to look at an alternative.

**Comment by wooorm** [comment]
Leaving the node algorithm here: https://nodejs.org/dist/latest-v14.x/docs/api/esm.html#esm_resolver_algorithm_specification

EDIT and: https://github.com/daKmoR/import-meta-resolve/blob/master/src/importMetaResolve.js

File: index.js (hunk 2 above)
Comment (wooorm) [reply index.js:12]: Could also be that `libnpmconfig` will swap that out for another project?

**Comment by wooorm** [comment]
I’m now the owner of `import-meta-resolve`, and will try to port how Node does that resolving. Probably **tons** of tests though, so that’ll take some time

**Comment by wooorm** [comment]
it works!

**Pull request closed without merging**